packages/fxa-settings/src/components/Settings/FlowSetupRecoveryPhoneSubmitNumber/index.tsx

```diff
@@ -0,0 +1,111 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import FlowContainer from '../FlowContainer';
+import ProgressBar from '../ProgressBar';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import Banner from '../../Banner';
+import { BackupRecoveryPhoneImage } from '../../images';
+import { getLocalizedErrorMessage } from '../../../lib/error-utils';
+import { useFtlMsgResolver } from '../../../models';
+import FormPhoneNumber from '../../FormPhoneNumber';
+
+export type FlowSetupRecoveryPhoneSubmitNumberProps = {
+  localizedBackButtonTitle: string;
+  localizedPageTitle: string;
+  navigateBackward: () => void;
+  navigateForward: () => void;
+  verifyPhoneNumber: (phoneNumber: string) => Promise<void>;
+};
+
+export const FlowSetupRecoveryPhoneSubmitNumber = ({
+  localizedBackButtonTitle,
+  localizedPageTitle,
+  navigateBackward,
+  navigateForward,
+  verifyPhoneNumber,
+}: FlowSetupRecoveryPhoneSubmitNumberProps) => {
+  const [localizedErrorBannerMessage, setLocalizedErrorBannerMessage] =
+    useState('');
+
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  const clearBanners = async () => {
+    return new Promise<void>((resolve) => {
+      setLocalizedErrorBannerMessage(''); // Clear the banner message
+      setTimeout(() => resolve(), 0); // Wait for DOM update
+    });
+  };
+
+  const handlePhoneNumber = async (phoneNumber: string) => {
+    await clearBanners();
+    try {
+      await verifyPhoneNumber(phoneNumber);
+      navigateForward();
+      return { hasErrors: false };
+    } catch (error) {
+      const localizedError = getLocalizedErrorMessage(ftlMsgResolver, error);
+      setLocalizedErrorBannerMessage(localizedError);
+      return { hasErrors: true };
+    }
+  };
+
+  return (
+    <FlowContainer
+      {...{ localizedBackButtonTitle }}
+      title={localizedPageTitle}
+      onBackButtonClick={navigateBackward}
+    >
+      <ProgressBar currentStep={1} numberOfSteps={2} />
```

Comment:
Feels like maybe you could lift/share that `numberOfSteps` const you set with this somehow, since if that changes this also will change.